### PR TITLE
Improvements in config.md file

### PR DESCRIPTION
**Pull request by prankul88**
**What this PR does / why we need it**:
It is related to #378 . 

**Special notes for your reviewer**:
I am a new contributor and faced these issues while setting up my development environment.
Wanted feedback regarding command `openstack security group rule create --egress` . I feel we must add `protocol` tag in this command.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -282,7 +282,7 @@ Instead of tagging, you also have the option to add metadata to instances. This
...
 
 ## Timeout settings
-During some heavy workload cloud, the time for create and delete openstack instance might takes long time, by default it's 5 minute.
-you can set:
-`CLUSTER_API_OPENSTACK_INSTANCE_DELETE_TIMEOUT` for instance delete timeout value.
-`CLUSTER_API_OPENSTACK_INSTANCE_CREATE_TIMEOUT` for instance create timeout value.
+During some heavy workload cloud, the time to create and delete openstack instance might take long time, by default it's 5 minute.
+You can set:
+`TimeoutInstanceDelete` for instance delete timeout value and
+`TimeoutInstanceCreate` for instance create timeout value in file `pkg/cloud/openstack/machine/actuator.go`
```
Comment (jichenjc) [inline docs/config.md]: others looks good, except htis one are you saying we need change the code in order to get the time out value? 
I think originally the code is trying to use environment variable so no code change @prankul88

Comment (prankul88) [reply docs/config.md]: Ohh okay. Will revert this change. Thanks

File: docs/config.md [hunk 2]
```diff
@@ -75,20 +75,20 @@ Your kubernetes cluster must be reachable from wherever cluster-api-provider-ope
...
 
-For the installer to work, a few security groups are required to be open. These may be different from the security groups needed to reach a cluster once its running. The following security group rules should be added to the security group of your chosing. For this example, we will suppose you created a security group names ``kubernetes`` that you will use for the cluster.
+For the installer to work, a few security groups are required to be open. These may be different from the security groups needed to reach a cluster once its running. The following security group rules should be added to the security group of your chosing. For this example, we will suppose you created a security group named ``kubernetes`` that you will use for the cluster.
 
 ```bash
 openstack security group rule create --ingress --protocol tcp --dst-port 22 kubernetes
 openstack security group rule create --ingress --protocol tcp --dst-port 3000:32767 kubernetes
 openstack security group rule create --ingress --protocol tcp --dst-port 443 kubernetes
-openstack security group rule create --egress
+openstack security group rule create --egress kubernetes
```
Comment (prankul88) [inline docs/config.md]: @jichenjc Wanted a feedback on this . As I mentioned in the description of this PR ,I did not get why there are no protocols for this egress rule.

Comment (jichenjc) [reply docs/config.md]: I am not sure, this was added by @iamemilio ,maybe we can get info from..

and if it's just 'kubenetes' missing, I assume that's a typo..., anyway, we can confirm from @iamemilio

Comment (prankul88) [reply docs/config.md]: Okay I will wait for his reply and then push a patchset for the changes.

File: docs/config.md (hunk 1 above)
Comment (prankul88) [reply docs/config.md]: Hi @jichenjc 
Kindly check, I have updated the patch.

File: docs/config.md (hunk 2 above)
Comment (prankul88) [reply docs/config.md]: I think this change is valid as network name is a positional argument in this command.
Will update if further chnage required

**Comment by jichenjc** [comment]
thanks for the update :)

**Pull request merged**